### ISSUE TITLE
[Bug] Tab position announcement is not preserved after the "select tabs" button becomes visible.

### ISSUE THREAD

**Issue by mcarare** [mozilla-mobile/fenix#16511]
In https://github.com/mozilla-mobile/fenix/pull/16481 we used an `accessibilityDelegate` to remove the "select tabs" button from the collection node info of the tabs list. 
Because we had to create a custom node info ( due to the use of `ConcatAdapter` that breaks a11y for the recycler) we also used an `accessibilityDelegate` for the same view.

The problem that arose from these separate fixes is that a `View` can have at most one `accessibilityDelegate`, so adding a second one breaks the functionality implemented by the previous one. In this case, when the list is loaded the first time (with the button not visible) the a11y works fine for the items, with position announcement within the list. But after the button is visible and the removal from the list occurs (by using the second `accessibilityDelegate`) we are left with the initial problem of a11y broken for the list. 

The most straightforward fix would be just leaving the button in the collection node and just removing the item info, so the issue described in https://github.com/mozilla-mobile/fenix/issues/16416 will not occur, and Talkback will not announce "Save to collection, Button, 3 of 3", but just "Save to collection". The only downside would be that Talkback would still consider it part of the list.

**Comment by mcarare** [comment]
Note: The fix for this particular problem was implemented without tackling the underlying problem of multiple a11y delegates. The QA for the fix will be made in issue https://github.com/mozilla-mobile/fenix/issues/16416.

The issue of needing to have multiple a11y delegates should however be handled and a general solution should be implemented. I've opened an issue for this problem that would eventually be technical debt if further a11y enhancements should be implemented on multiple custom screens we have in our app: https://github.com/mozilla-mobile/fenix/issues/16556.

**Comment by mcarare** [comment]
Closing as the initial issue is marked as fixed: https://github.com/mozilla-mobile/fenix/issues/16416#issuecomment-728811347.
